initializer/mount.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// TODO(jmxnzo) don't hardcode these, add struct EncryptedMount as parameter to setupEncryptedMount function
+	// TODO follow naming convention that /dev/mapper/{x} is remounted to /{x}
+	// csiDevicePath is the path to the block device to be encrypted.
+	csiDevicePath = "/dev/csi0"
+
+	mapperDeviceMountPoint = "state"
+	// workloadSecretPath is the path to the Contrast workload secret.
+	workloadSecretPath = "/contrast/secrets/workload-secret-seed"
+
+
+	// tmp_key_path is the path to a temporary key file.
+	tmpKeyPath = "/dev/shm/key"
+	// disk_encryption_key_path is the path to the disk encryption key.
+	diskEncryptionKeyPath = "/dev/shm/disk-key"
+)
+
+// SetupEncryptedMount sets up an encrypted LUKS volume mount using the specification of struct EncryptedMount handed in.
+func setupEncryptedMount(ctx context.Context, logger *slog.Logger) error {
+	if !isLuks(ctx, logger, csiDevicePath) {
+		// TODO(jmxnzo) might just use stdin instead for the initial passphrase generation
+		if err := createInitPassphrase(tmpKeyPath); err != nil {
+			return err
+		}
+		logger.Info("formatting csi device to LUKS with initial passphrase)")
+		if err := luksFormat(ctx, csiDevicePath, tmpKeyPath); err != nil {
+			return err
+		}
+
```

Comment:
We should consider what happens if the container is terminated in between formatting and changing the key. At least add a TODO for now.